modules/nf-core/samtools/bam2fq/main.nf

```diff
@@ -4,12 +4,12 @@ process SAMTOOLS_BAM2FQ {
 
     conda "${moduleDir}/environment.yml"
     container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
-        'https://depot.galaxyproject.org/singularity/samtools:1.17--h00cdaf9_0' :
-        'biocontainers/samtools:1.17--h00cdaf9_0' }"
+        'https://depot.galaxyproject.org/singularity/samtools:1.19.2--h50ea8bc_0' :
+        'biocontainers/samtools:1.19.2--h50ea8bc_0' }"
 
     input:
     tuple val(meta), path(inputbam)
-    val split
+    val(split)
```

Comment:
if I understand correctly this is `true` if paired end, `false` if single_end right? that will just be stored in the `meta` field so is it really necessary to have this input? I noticed it complicates calling the function later in `decontamination.nf` (see later comment)

Reply:
I would do  what you did with other modules and just have a `def single_end` using the `meta.single_end`